plugins/jira-dashboard-backend/src/service/router.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  CacheManager,
+  TokenManager,
+  errorHandler,
+} from '@backstage/backend-common';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+import { CatalogClient } from '@backstage/catalog-client';
+import { DiscoveryApi } from '@backstage/plugin-permission-common';
+import { IdentityApi } from '@backstage/plugin-auth-node';
+import {
+  getFiltersFromAnnotations,
+  getIssuesFromComponents,
+  getIssuesFromFilters,
+  getProjectResponse,
+} from './service';
+import { getDefaultFilters } from '../filters';
+import {
+  COMPONENT_ANNOTATION,
+  FILTER_ANNOTATION,
+  Filter,
+  JiraResponse,
+  PROJECT_KEY_ANNOTATION,
+  Project,
+} from '@backstage/plugin-jira-dashboard-common';
+import { getProjectAvatar } from '../api';
+import stream from 'stream';
+
+/**
+ * @public
+ */
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+  discovery: DiscoveryApi;
+  identity: IdentityApi;
+  tokenManager: TokenManager;
+}
+
+const DEFAULT_TTL = 1000 * 60;
+
+/**
+ * @public
+ */
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger, config, discovery, identity, tokenManager } = options;
+  const catalogClient = new CatalogClient({ discoveryApi: discovery });
+  logger.info('Initializing Jira Dashboard backend');
+
+  const pluginCache =
+    CacheManager.fromConfig(config).forPlugin('jira-dashboard');
+  const cache = pluginCache.getClient({ defaultTtl: DEFAULT_TTL });
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/health', (_, response) => {
+    response.json({ status: 'ok' });
+  });
+
+  router.get('/:entityRef', async (request, response) => {
```

Comment:
Just a hint that if you add a ROOT route that's parameterized like this (no prefix at all), you will be severely locked down in your ability to iterate on the API in the future without having complex collisions to deal with. You may want to add like `/dashboards/by-entity-ref/` or something as a prefix on this, whichever makes sense

Reply:
I didn't know that but good to learn. Will fix it now!